src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/ReferralEventPublisher.kt

```diff
@@ -38,6 +47,19 @@ class ReferralEventPublisher(
     applicationEventPublisher.publishEvent(ReferralEvent(this, eventType, referral, getSentReferralURL(referral)))
   }
 
+  fun referralDetailsChangedEvent(referral: Referral, newDetails: ReferralDetails, previousDetails: ReferralDetails) {
+    applicationEventPublisher.publishEvent(
+      ReferralEvent(
+        this, ReferralEventType.DETAILS_CHANGED, referral, getSentReferralURL(referral),
+        mapOf(
```

Comment:
using DTOs here to avoid having to do any entity related ops in the async listener - avoids 'failed to lazily initialize a collection of role' errors